packages/astro/src/cli/info/index.ts

```diff
@@ -12,31 +15,82 @@ interface InfoOptions {
 	flags: Flags;
 }
 
+export const VERSION_PREFIX_REGEX = /[\^~<>=]/g;
+
+type MinimalPackageJSON = {
+	version: string;
+	dependencies: Record<string, string>;
+}
+
+function getVersion(packageJsonPath: string, dependency?: string): string | undefined {
+	try {
+		const resolvedPath = createRequire(import.meta.url).resolve(packageJsonPath);
+		
+		const packageJSON = JSON.parse(readFileSync(resolvedPath, "utf-8")) as MinimalPackageJSON;
+		
+		if (dependency) {
+			if (!packageJSON.dependencies) return undefined;
+			
+			const depVersion = packageJSON.dependencies[dependency];
+			
+			if (!depVersion) return undefined;
+			
+			const sanitizedDepVersion = depVersion.replaceAll(VERSION_PREFIX_REGEX, "");
+			
+			return sanitizedDepVersion;
+		} else {
+			return packageJSON.version;
+		}
+	} catch {
+		return undefined;
+	}
+}
+
 export async function getInfoOutput({
 	userConfig,
 	print,
+	root
 }: {
 	userConfig: AstroUserConfig | AstroConfig;
 	print: boolean;
+	root: string;
 }): Promise<string> {
+	const astroViteVersion = getVersion("../../../package.json", "vite");
+	const userViteVersion = getVersion(path.join(root, "package.json"), "vite");
+	
 	const rows: Array<[string, string | string[]]> = [
 		['Astro', `v${ASTRO_VERSION}`],
+		['Vite', userViteVersion ? `User: v${userViteVersion}, Astro: ${astroViteVersion}` : `Astro: v${astroViteVersion}`],
 		['Node', process.version],
 		['System', getSystem()],
 		['Package Manager', getPackageManager()],
 	];
-
+	
+	const adapterVersion = userConfig.adapter?.name
+		? getVersion(userConfig.adapter.name)
+		: undefined;
+	
+	const adatperOutputString = userConfig.adapter?.name
+		? `${userConfig.adapter.name} ${adapterVersion ? `(v${adapterVersion})` : ""}`
+		: "none";
+	
 	try {
 		rows.push(['Output', userConfig.output ?? 'static']);
-		rows.push(['Adapter', userConfig.adapter?.name ?? 'none']);
+		rows.push(['Adapter', adatperOutputString]);
 		const integrations = (userConfig?.integrations ?? [])
 			.filter(Boolean)
 			.flat()
-			.map((i: any) => i?.name)
+			.map((i: any) => {
+				const name = i?.name;
+				const version = getVersion(`${name}/package.json`);
```

Comment:
This has two assumptions that I don't think are safe:
- That integrations have their `name` property set to the package name (official ones all do, but not community ones)
- That integrations expose a `package.json` export (which they might not, since this is ESM)